bitmap.go

```diff
@@ -189,11 +189,12 @@ func (b *Bitmap) createSegmentIfNotExists(slice uint64) *BitmapSegment {
 	}
 
 	// Insert new segment.
-	b.segments = append(b.segments, BitmapSegment{})
+	b.segments = append(b.segments, BitmapSegment{data: *roaring.NewBitmap()})
 	if i < len(b.segments) {
 		copy(b.segments[i+1:], b.segments[i:])
 	}
 	b.segments[i] = BitmapSegment{
+		data:     *roaring.NewBitmap(),
```

Comment:
change this to NewBTreeBitmap?

Reply:
no, in this instance I think it will be  better to use the slice base bitmap since the user of segments will be append only.

Reply:
are you sure? It looks like `createSegmentIfNotExists` is used by `SetBit` which is used by `Fragment.SetBit`

Reply:
my current running theory is that only what I am calling file based roaring bitmaps should be btree backed.  I think what I col row based bitmaps should be sliced backed(for they have very few containers relative to the file-backed ones.

Reply:
I see, that makes sense. Whether or not that's worth the increased complexity is still an open question to me though

Reply:
benchmarks would indicated it was imho

Reply:
👍